docs/threat-model.md

```diff
@@ -0,0 +1,56 @@
+# Threat model
+This document briefly describes the threat model of this repository.
+
+The `team` repository manages various sensitive permissions:
+- It provides access to GitHub users to rust-lang (and other organization's) repositories.
+    - This includes `admin` access, which can be quite destructive.
+- It provides access to private Zulip streams.
+- It subscribes users to mailing lists.
+
+## User categories
+There are three groups of users that come into play here:
+- Infra *admins*. This is a small group of users that essentially have all the permissions.
+- Team repo *maintainers*. This is a larger set of users with `write` access to this repository, which can include e.g. team leads.
+- *Unprivileged* users, which is everyone else. This includes people that send PRs to this repository.
+
+On the one hand, we want to give benevolent access to team leads to merge "benign" changes, such as membership changes in teams without elevated privileges or modifications to the description or homepage URL of repositories. On the other hand, we must make sure that the maintainers cannot make more dangerous changes, such as giving accessing CI tokens (more on that below). Otherwise, their accounts could become a target.
+
+## What we need to avoid
+After a `team` PR is approved, it is put into a merge queue, which uses a GitHub environment to access a secret "root" token that can do pretty much anything across the rust-lang enterprise. This secret is then used to sync changes to GitHub/Zulip/Mailgun, based on the changes made in the PR.
```

Comment:
```suggestion
After a `team` PR is approved, it is put into a merge queue, which uses a GitHub environment to access secret tokens that can do pretty much anything across the rust-lang enterprise. These secrets are then used to sync changes to GitHub/Zulip/Mailgun, based on the changes made in the PR.
```
repetition. Also, it might be confusing what root means here. in the future this token will be a github app.
Also github, zulip, and mailgun have different tokens